pandas/types/concat.py

```diff
@@ -297,6 +300,9 @@ def _maybe_unwrap(x):
         else:
             raise TypeError('Categorical.ordered must be the same')
 
+    if ignore_order:
+        ordered = False
```

Comment:
The if statement on line 264 can be entered if the ordered categoricals have the same categories and order.

is_dtype_equal checks categories and ordering